src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
+		name             string
+		testFn           testFn
+		shouldError      bool
+		expectedErrorMsg string
+		expectedResponse interface{}
+	}
+)
+
+func TestStartAutomatedRecovery(t *testing.T) {
+	successfullStartResponse := RecoveryStartResponse{
+		Code:    200,
+		Message: "Automated recovery started",
+	}
+
+	ctx := context.Background()
```

Comment:
[nit] No need to create a variable since it's only used bellow for `client.StartAutomatedRecovery(ctx, ".*")`

And same in the other test.